src/gui/src/gui.cpp

```diff
@@ -1338,10 +1338,10 @@ int startGui(int& argc,
       &app, &QApplication::aboutToQuit, main_window, &MainWindow::saveSettings);
 
   // execute commands to restore state of gui
-  std::string restore_commands;
-  for (const auto& cmd : gui->getRestoreStateCommands()) {
-    restore_commands += cmd + "\n";
-  }
+  std::vector<std::string> restore_commands;
```

Comment:
I don't see any reason why the restore commands need any access to script info and therefore there should not be any changes in this. There is no CI checks for this, so this needs to be tested manually.

Reply:
You're right that there are no tests at all for `openroad -gui`, seems like that should be remedied.

We already peer into the restore commands(to look for exit), so storing them as separate commands leaves it to the backend(during userinterface) to inspect the commands and give them(source & exit) any special treatment.

Reply:
Aha! That was surprising. Any idea why "source" has been overriden instead of keeping an OpenSTA specific name?